plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2016, Peter Sagerson <psagers@ignorare.net>
+# Copyright: (c) 2020, Sebastian Pfahl <eryx@gmx.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r"""
+---
+module: ldap_search
+short_description: Search for entries in a LDAP server.
+description:
+  - Return the results of an LDAP search. Use Ansible's 'register' statement.
+notes:
+  - Only supports Python3
+  - The default authentication settings will attempt to use a SASL EXTERNAL
+    bind over a UNIX domain socket. This works well with the default Ubuntu
+    install for example, which includes a cn=peercred,cn=external,cn=auth ACL
+    rule allowing root to modify the server configuration. If you need to use
+    a simple bind to access your server, pass the credentials in I(bind_dn)
+    and I(bind_pw).
+version_added: "2.9"
+author:
+  - Sebastian Pfahl (@eryx12o45)
+requirements:
+  - python-ldap
+options:
+  dn:
+    required: True
+    type: str
+    description:
+      - The LDAP DN to search in.
+  scope:
+    choices: [base, onelevel, subordinate, children]
+    default: base
+    type: str
+    description:
+      - The LDAP scope to use.
+  filter:
+    default: '(objectClass=*)'
+    type: str
+    description:
+      - Used for filtering the LDAP search result.
+  attrs:
+    type: raw
```

Comment:
```suggestion
    type: list
    elements: str
```
Ansible automatically converts `attr: a,b,c` to `attr: [a, b, c]` if you simply specify a comma-separated string